src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class FailOverSupportedConfigurationClient : IConfigurationClient
+    {
+        private const int HttpStatusRequestThrottled = 429;
+
+        private readonly IEnumerable<LocalConfigurationClient> clients;
+
+        private DateTimeOffset retryPrimaryConfigClientAfter;
+
+        private int failedRequestsToPrimaryConfigClient = 0;
+
+        public FailOverSupportedConfigurationClient(IEnumerable<LocalConfigurationClient> clients)
```

Comment:
How about just taking the args to create configurationclients itself rather than having a new ConfigurationClient type.

Reply:
The reason for having `LocalConfigurationClient` is in the comment above. If we can solve that problem, we can get rid of the `LocalConfigurationClient`.